x/posts/internal/keeper/invariants.go

```diff
@@ -0,0 +1,130 @@
+func AllInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		if res, stop := Hash256PostIDInvariant(k)(ctx); stop {
+			return res, stop
+		}
+
+		if res, stop := ValidCommentsDateInvariants(k)(ctx); stop {
+			return res, stop
+		}
+
+		if res, stop := ValidPollForPollAnswersInvariants(k)(ctx); stop {
+			return res, stop
+		}
+
+		if res, stop := ValidPostForReactionsInvariants(k)(ctx); stop {
+			return res, stop
+		}
+
+		return "Every invariant condition is fulfilled correctly", true
+	}
+}
+
+// formatOutputIDs concatenate the ids given into a unique string
+func formatOutputIDs(ids types.PostIDs) (outputIDs string) {
+	for _, id := range ids {
+		outputIDs += id.String() + "\n"
+	}
+	return outputIDs
+}
+
+// Hash256PostIDInvariant checks that the all posts have a SHA256 ID
+func Hash256PostIDInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var invalidPostIDs types.PostIDs
+		k.IteratePosts(ctx, func(_ int64, post types.Post) (stop bool) {
+			if !post.PostID.Valid() {
+				invalidPostIDs = append(invalidPostIDs, post.PostID)
+			}
+			return false
+		})
+
+		return sdk.FormatInvariant(types.ModuleName, "invalid post IDs",
+			fmt.Sprintf("The following post IDs are invalid:\n %s", formatOutputIDs(invalidPostIDs)),
+		), invalidPostIDs != nil
+	}
+}
+
+// ValidCommentsDateInvariants checks that comments creation date is always greater than parent creation date
+func ValidCommentsDateInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var invalidCommentsIDs types.PostIDs
+		k.IteratePosts(ctx, func(_ int64, post types.Post) (stop bool) {
+			if post.ParentID.Valid() {
+				parentPost, _ := k.GetPost(ctx, post.ParentID)
+				if post.Created.Before(parentPost.Created) {
+					invalidCommentsIDs = append(invalidCommentsIDs, post.PostID)
+				}
+			}
+			return false
+		})
+
+		return sdk.FormatInvariant(types.ModuleName, "comments dates before parent post date",
+			fmt.Sprintf("The following post IDs referred to posts which are invalid comments "+
+				"having creation date before parent post creation date:\n %s",
+				formatOutputIDs(invalidCommentsIDs)),
+		), invalidCommentsIDs != nil
+	}
+}
+
+// ValidPostForReactionsInvariants checks that the post related to the reactions is valid and exists
+func ValidPostForReactionsInvariants(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var invalidPostIDs types.PostIDs
+		reactions := k.GetReactions(ctx)
+		for key := range reactions {
+			postID := types.PostID(key)
+			if _, found := k.GetPost(ctx, postID); !found {
+				invalidPostIDs = append(invalidPostIDs, postID)
+			}
+		}
+
+		return sdk.FormatInvariant(types.ModuleName, "posts reactions refers to non existing posts",
+			fmt.Sprintf("The following post IDs referred to posts that did not exist:\n %s",
+				formatOutputIDs(invalidPostIDs)),
+		), invalidPostIDs != nil
```

Comment:
I think that probably, instead of using a list of `PostIDs` we might want to use a list of `Reaction` instead. What do you think? That way it could be easier to understand what's wrong. 

Also, on the message we should probably tell "The following reactions refer to posts that do not exist".